doc/source/user/numpy-for-matlab-users.rst

```diff
@@ -25,165 +25,51 @@ Some Key Differences
 
 .. list-table::
 
-   * - In MATLAB®, the basic data type is a multidimensional array of
-       double precision floating point numbers.  Most expressions take such
-       arrays and return such arrays.  Operations on the 2-D instances of
-       these arrays are designed to act more or less like matrix operations
-       in linear algebra.
-     - In NumPy the basic type is a multidimensional ``array``.  Operations
-       on these arrays in all dimensionalities including 2D are element-wise
-       operations.  One needs to use specific functions for linear algebra
-       (though for matrix multiplication, one can use the ``@`` operator
-       in python 3.5 and above).
-
-   * - MATLAB® uses 1 (one) based indexing. The initial element of a
+   * - In MATLAB, the basic type is a multidimensional array. By default, MATLAB
+       creates two dimensional arrays of double precision floating point numbers.
+       Operations on the 2D instances of these arrays are designed to act like
+       matrix operations in linear algebra. Most expressions take such arrays and
+       return arrays.
+     - In NumPy, the basic type is a multidimensional ``array``. By default,
+       NumPy creates 0-D or 1-D arrays for inputs of numbers or lists,
+       respectively.  Operations on these arrays in all dimensionalities
+       including 2D are element-wise operations.  One needs to use specific
+       functions for linear algebra (though for matrix multiplication, one can
+       use the ``@`` operator in Python 3.5 and above).
+
+   * - MATLAB uses 1 (one) based indexing. The initial element of a
        sequence is found using a(1).
```

Comment:
I like this change, but I made one small edit
`MATLAB numbers indexes from 1; a(1) is the first element.`
to
`MATLAB numbers indices from 1; a(1) is the first element.`

Reply:
Thanks for making the change.

The [Google style guide](https://developers.google.com/style/word-list#letter-i) has a rule for the plural of **index**:

> **index**
>     Use the plural indexes unless there is a domain-specific reason (for example, a mathematical or financial context) to use 

I think this is an "indexes" situation; calling it a mathematical context is a stretch. 

If our docs uniformly used "indices" we'd toss the rule, but that's not the case.

The two or three other "indices" instances on the page, which I overlooked, ought to be changed as well.

Reply:
Given numpy has a function `np.indices`, I think we should use that spelling everywhere.

Reply:
> Given numpy has a function `np.indices`, I think we should use that spelling everywhere.

@bjnath, Thanks for the Google style guide. I think keeping the indices and `np.indices` would be nice to keep consistent with the numpy function. I would classify this as mathematical context, because we are discussing arrays and matrices. 

After this discussion, I noticed one other "indexes" -> "indices" on L580 in the __Notes__. I'll push the change with next commit.

Reply:
I loathe "indices" but others equally loathe "indexes," and I have no good counter for `np.indices`, which I guess goes back as far as one can go.  

I'll find a place to collect style rules that supplement or override Google's.